src/crimson/os/seastore/seastore_types.h

```diff
@@ -401,35 +500,54 @@ enum class addr_types_t : uint8_t {
   RANDOM_BLOCK = 1
 };
 struct seg_paddr_t;
+struct blk_paddr_t;
 struct paddr_t {
 protected:
   using common_addr_t = uint64_t;
   common_addr_t dev_addr;
 private:
-  constexpr paddr_t(segment_id_t seg, segment_off_t offset)
+  constexpr paddr_t(segment_id_t seg, seastore_off_t offset)
     : dev_addr((static_cast<common_addr_t>(seg.segment)
 	<< SEG_OFF_LEN_BITS) | static_cast<uint32_t>(offset)) {}
   constexpr paddr_t(common_addr_t val) : dev_addr(val) {}
+  constexpr paddr_t(block_id_t blk, seastore_off_t offset)
```

Comment:
block_id_t contains an offset, doesn't it?  Why isn't this device_id_t, seastore_off_t?

Reply:
It seems that we need to discuss blk_paddr_t format. I added a comment as below.